File: Week1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void) 
{
    // initialazing variables
    int steps = 0;
    float cash;
    
    // asking for user input in dollars
    do
    {
        printf("How much cash is owed to the customer? ");
        cash = get_float();
    }
    while (cash < 0.00);
    
    // converting dollars to cents
    cash = (int)round(cash*100);
    
    // giving the change back in the most efficient way
    do
    {
        if (cash >= 25) 
        {
            cash -= 25;
            steps++;
        }
        else if (cash >= 10) 
        {
            cash -= 10;
            steps++;
        }
        else if (cash >= 5) 
        {
            cash -= 5;
            steps++;
        }
        else if (cash >= 1) 
        {
            cash -= 1;
            steps++;
        }
    }
    while (cash > 0);
    
    // says how many coins we used to give the change back
    printf("%d\n",steps);
    return 0;
}

File: pset6/greedy.py
# branko stankovic
# cs 50 edX course
# https://branko-stankovic.github.io/

# cs50 library
import cs50

# forever loop, until user provides a proper input
while True:
    print("How much cash is owed to the customer?")
    cash = cs50.get_float()
    if cash > 0.00:
        break

# initialize a number of steps to return the change
steps = 0

# rounding and multiplying by 100 to get integers instead of floating point
cash = int(round(cash*100))

# first return quarters
while cash >= 25:
    cash -= 25
    steps += 1

# return dimes    
while cash >= 10:
    cash -= 10
    steps += 1

# return nickel
while cash >= 5:
    cash -= 5
    steps += 1

# return pennies    
while cash >= 1:
     cash -= 1
     steps += 1

# how much steps it took to return the change?     
print(steps)     